src/init.cpp

```diff
@@ -1566,7 +1566,7 @@ bool AppInitMain(NodeContext& node)
                 // It both disconnects blocks based on ::ChainActive(), and drops block data in
                 // BlockIndex() based on lack of available witness data.
                 uiInterface.InitMessage(_("Rewinding blocks...").translated);
-                if (!RewindBlockIndex(chainparams)) {
+                if (!::ChainstateActive().RewindBlockIndex(chainparams)) {
```

Comment:
175e570

I'm not sure if `RewindBlockIndex` is still relevant.

Following Suhas comment https://github.com/bitcoin/bitcoin/pull/8149#issuecomment-257943026, it was added to let upgrading nodes after segwit activation redownload consensus data without redownloading the whole blockchain.

According to sipa in #15402, the rewinding logic "is probably becoming increasingly unnecessary, as very few pre-0.13.1 nodes remain that would care to upgrade".

Further, it doesn't make sense for the assumed-valid UTXO chainstate to rewind until first insufficiently-validated block, all utxo hash are going to be post segwit. It only makes sense for the ibd chainstate.

Reply:
Agree this is worth investigating, but probably not within this PR. I'd rather just preserve existing behavior.

Reply:
FYI, https://github.com/bitcoin/bitcoin/issues/17862. I had a second look, I think it's safe to maintain logic as it is for this PR even it should be never triggered because both chainstates will belong to segwit nodes.